recipes/xtr/all/conanfile.py

```diff
@@ -1,125 +1,159 @@
-from conans import ConanFile, AutoToolsBuildEnvironment, tools
-from conans.errors import ConanInvalidConfiguration
-
 import os
 
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMakeToolchain, CMakeDeps, CMake
+from conan.tools.files import copy, get, replace_in_file
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.53.0"
+
 
 class XtrConan(ConanFile):
     name = "xtr"
-    description = \
-        "C++ Logging Library for Low-latency or Real-time Environments"
-    topics = ("xtr", "logging", "logger")
+    description = "C++ Logging Library for Low-latency or Real-time Environments"
+    license = "MIT"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/choll/xtr"
-    license = "MIT"
+    topics = ("logging", "logger")
+
+    package_type = "library"
     settings = "os", "arch", "compiler", "build_type"
     options = {
+        "shared": [True, False],
         "fPIC": [True, False],
         "enable_exceptions": [True, False],
         "enable_lto": [True, False],
         "enable_io_uring": ["auto", True, False],
         "enable_io_uring_sqpoll": [True, False],
-        "sink_capacity_kb": "ANY"
+        "sink_capacity_kb": [None, "ANY"],
     }
     default_options = {
+        "shared": False,
         "fPIC": True,
         "enable_exceptions": True,
         "enable_lto": False,
         "enable_io_uring": "auto",
         "enable_io_uring_sqpoll": False,
-        "sink_capacity_kb": None
+        "sink_capacity_kb": None,
     }
 
-    generators = "make"
-
     def config_options(self):
-        if tools.Version(self.version) < "1.0.1":
-            del self.options.sink_capacity_kb
-        if tools.Version(self.version) < "2.0.0":
+        if Version(self.version) < "2.0.0":
             del self.options.enable_io_uring
             del self.options.enable_io_uring_sqpoll
 
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
```

Comment:
Surely you'd want fPIC enabled for a shared library?

Reply:
Yup, this disables the fPIC option, making it non-optional. It's standard recipe boilerplate on CCI: https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/conanfile.py#L61-L67